afs-storage-api/src/main/java/com/powsybl/afs/storage/AppStorageDataStore.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.afs.storage;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.powsybl.commons.datasource.DataSource;
+import com.powsybl.commons.datastore.DataStore;
+
+/**
+ * A datastore corresponding to a data blob stored in the file system.
+ * A data blob is associated to a node and a name identifying it among data blobs of this node.
+ *
+ * @author Giovanni Ferrari <giovanni.ferrari at techrain.eu>
+ */
+public class AppStorageDataStore implements DataStore {
+
+    private static final String SEPARATOR = "__";
+
+    public interface Name {
```

Comment:
It seems that the implementation is a bit complicated for what we need. 
Maybe it was needed for the "data source" in order to handle the suffix and extensions, but I think we should do something simpler here.

I think that here all we need is a naming convention like "DATA_STORE_ENTRY_<actual entry name>", + a simple method `String getDataName(String entryName)`. 

The day we want to allow a different mapping entry name --> data name, we could pass such a function to the constructor. But for the moment it does not seem useful.